zokrates_core/src/semantics.rs

```diff
@@ -1263,7 +1372,58 @@ impl<'ast, T: Field> Checker<'ast, T> {
                             }),
                         }
                     }
-                    _ => unreachable!("user defined types should always be structs"),
+                    (declaration_type, Some(alias_generics)) => {
+                        match alias_generics.len() == generics.len() {
+                            true => {
+                                let generic_identifiers =
+                                    alias_generics.iter().map(|c| match c.as_ref().unwrap() {
+                                        DeclarationConstant::Generic(g) => g.clone(),
+                                        _ => unreachable!(),
+                                    });
+
+                                // build the generic assignment for this type
+                                let assignment = GGenericsAssignment(generics
+                                    .into_iter()
+                                    .zip(generic_identifiers)
+                                    .map(|(e, g)| match e {
+                                        Some(e) => {
+                                            self
+                                                .check_expression(e, module_id, types)
+                                                .and_then(|e| {
+                                                    UExpression::try_from_typed(e, &UBitwidth::B32)
+                                                        .map(|e| (g, e))
+                                                        .map_err(|e| ErrorInner {
+                                                            pos: Some(pos),
+                                                            message: format!("Expected u32 expression, but got expression of type {}", e.get_type()),
+                                                        })
+                                                })
+                                        },
+                                        None => Err(ErrorInner {
```

Comment:
I don't think this should be allowed at all in alias definitions: all generics must always be passed. Maybe the error message is incorrect, or this is actually unreachable depending on what happens in the parser (maybe we create Some earlier and unwrap here, never creating None so this wiould be unreachable)

Reply:
Oh this is for Type not DeclarationType so `_` is allowed.